client/src/components/NationwideMap.vue

```diff
@@ -323,20 +323,20 @@ export default defineComponent({
     },
     // Listens to query param to toggle layers.
     routerLayer: function(newLayer: string) {
-      if (newLayer != null) {
+      if (newLayer != null && this.map?.isStyleLoaded()) {
         this.setDataLayerVisibility(newLayer, true);
       }
     },
     // Listen for changes to lat/long to update map location.
     'geoState.geoids': function() {
+      if (this.marker) {
+        this.marker.remove();
```

Comment:
You don't know that this marker exists yet do you?

Reply:
oh this is checking if it exists I see!

Reply:
Changed to what you had anyway - it's more clear!